### ISSUE TITLE
Tag Matching Bug

### ISSUE THREAD

**Issue by pburleson** [nvie/gitflow#91]
There appears to be  a tag matching bug. I was trying to run the command: 

git flow hotfix start 1.0.2

It kept failing with: 

Tag '1.0.2' already exists. Pick another name.

Looking at the source, the list of current tags is provided by "git tag;"

My output for that command is: 

1.0
1.0.1
1.0b1
1.0b2
1.0b3
1.0b4
1.0b5

Once I deleted the 1.0b2 tag, I was able to create the hotfix branch with the name 1.0.2

The problem appears to be in gitflow-common for the function:

```
require_tag_absent() {
    if has $1 $(git_all_tags); then
        die "Tag '$1' already exists. Pick another name."
    fi
}
```

the "if has $1 $(git_all_tags); then" line is the issue. My shell scripting is a little rusty, but I'm not sure why 1.0.2 is matching 1.0b2.

**Comment by vladdu** [comment]
Hi,
Could it be that the dots should be escaped so that they aren't interpreted as control chars in the regexp?
regards,
Vlad

**Comment by pburleson** [comment]
Vlad, you're right. The . in the versions are matching "any character". I'll see if I can patch in to make the grep do a literal match.

**Comment by BinaryMuse** [comment]
Of interesting may be [`git-check-ref-format`](http://www.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html). It describes what characters are not allowed in a refname. Asterisks and plus- and dollar-signs are other characters that are valid in a tag name but may give bash/grep problems.

Played around with `sed` and something like

```
echo '$1.0.2+*' | sed 's/\([\.\+\$\*]\)/\\\1/g'
```

which results in

```
\$1\.0\.2\+\*
```

may be what you're looking for (or at least on the right track)? I'm not sure about the dollar sign issue as gitflow-common uses `echo " $@ " | grep -q " $item "` for its `has` function.

**Comment by nvie** [comment]
I've tried to fix this one in 1b471a66b43ee95067b53937c62dc42d4bc561e6. Please confirm that this solves your problem.
